internal/kubernetes/operator/project/project.go

```diff
@@ -545,49 +524,50 @@ func buildNetworkPeering(npProvider store.PeeringConnectionLister, projectID str
 	return result, nil
 }
 
-func convertNetworkPeer(np interface{}, providerName provider.ProviderName) atlasV1.NetworkPeer {
-	switch v := np.(type) {
-	case *atlasv2.AwsNetworkPeeringConnectionSettings:
-		return convertAWSNetworkPeer(v, providerName)
-	case *atlasv2.GCPNetworkPeeringConnectionSettings:
-		return convertGCPNetworkPeer(v, providerName)
-	case *atlasv2.AzureNetworkPeeringConnectionSettings:
-		return convertAzureNetworkPeer(v, providerName)
+func convertNetworkPeer(np atlasv2.BaseNetworkPeeringConnectionSettings, providerName provider.ProviderName) atlasV1.NetworkPeer {
+	switch np.GetProviderName() {
+	case "AWS":
+		return convertAWSNetworkPeer(&np, providerName)
+	case "GCP":
+		return convertGCPNetworkPeer(&np, providerName)
+	case "Azure":
+		return convertAzureNetworkPeer(&np, providerName)
+	default:
+		return atlasV1.NetworkPeer{}
```

Comment:
related to another comment, these switches allowed for type casting, they don't do that anymore and seems like safe to delete

Reply:
By removing those we will lose information of what consists an type.